File: fragments/solana_airdrop/solana_airdrop_utils.test.ts
import { getBalance } from "../solana_balance/solana_balance_utils";
import { sendAndConfirmAirdrop } from "./solana_airdrop_utils";
import assert from "node:assert/strict";
import test, { describe } from "node:test";
import { generateKeyPairSigner } from "@solana/kit";
import { LAMPORTS_PER_SOL } from "@solana/web3.js";

describe("solana airdrop utils", () => {
  test("sendAndConfirmAirdrop to new address", async () => {
    const keypair = await generateKeyPairSigner();
    const address = keypair.address;
    const balance = await getBalance(address);
    assert.strictEqual(balance, 0n);

    await sendAndConfirmAirdrop(address, BigInt(LAMPORTS_PER_SOL));
    const newBalance = await getBalance(address);
    assert.strictEqual(newBalance, BigInt(LAMPORTS_PER_SOL));
  });
});

File: fragments/solana_airdrop/solana_airdrop_utils_test.go
package solana_airdrop

import (
	"ff/solana_balance"
	"testing"

	"github.com/gagliardetto/solana-go"
)

func TestSolanaAirdrop(t *testing.T) {
	keypair, err := solana.NewRandomPrivateKey()
	if err != nil {
		t.Errorf("Failed to create keypair: %v", err)
	}

	balance, err := solana_balance.GetBalance(keypair.PublicKey())
	if err != nil {
		t.Errorf("Failed to get initial balance: %v", err)
	}

	if balance != 0 {
		t.Errorf("expected initial balance of zero but got: %d", balance)
	}

	err = SendAndConfirmAirdrop(keypair.PublicKey(), solana.LAMPORTS_PER_SOL)
	if err != nil {
		t.Errorf("SendAndConfirmAirdrop failed: %v", err)
	}

	latestBalance, err := solana_balance.GetBalance(keypair.PublicKey())
	if err != nil {
		t.Errorf("Failed to get latest balance %v", err)
	}

	if latestBalance != solana.LAMPORTS_PER_SOL {
		t.Errorf("Expected balance to be %d, but got %d", solana.LAMPORTS_PER_SOL, latestBalance)
	}
}
